### PR TITLE
Added Documentation

### PR DESCRIPTION
**Pull request by Carelvd**
The original repository did not provide for documentation.  While this may readily be included by invoking `sphinx-quickstart` not all users may know to do so. Furthermore there is a way to setup documentation such that it may be compiled directly through the `setup.py` file which even fewer people know about. 

This commit provides a documentation folder and the necessary information under `setup.cfg` for the final user to generate this documentation by invoking `python setup.py sphinx_build -b  html|singlehtml|latex|...`.

I'm afraid I'm not familiar enough with Cokkie Cutter to set up the dates to be automatically populated with the appropriate/current date and time. If you know how to o this please modify the Pr accordingly.

For more information please see the following links :

 -  `http://www.sphinx-doc.org/en/stable/setuptools.html`
 -  `https://docs.openstack.org/pbr/latest/user/using.html#build-sphinx`

### CODE REVIEW

**Comment by Carelvd** [comment]
Dang it,I forgot to add the following line to the readme and can't see how to add the commit to the pull request, please include the following if you accept the PR.

 - One may generate the documentation their project by running `python setup.py build_sphinx -b html|singlehtml|latex|...` from the project root.

**Comment by chintal** [comment]
+1

Having sphinx documentation stubs as part of the sphinxcontrib cookiecutter seems like a no brainer to me

**Comment by stephenfin** [comment]
@Carelvd Apologies for missing this. I wasn't subscribed to the repo, weirdly. Will take a look shortly!

File: README.rst [hunk 1]
```diff
@@ -32,14 +32,18 @@ Install `cookiecutter`::
 Generate a Python package::
 
     cookiecutter https://github.com/sphinx-contrib/cookiecutter.git
-
+
```
Comment (stephenfin) [inline README.rst:35]: Nit: can you drop this?

File: cookiecutter.json [hunk 2]
```diff
@@ -1,8 +1,16 @@
...
+  "short_description"    : "A short description of the extension",
+  "author_name"          : "Your name",
+  "author_email"         : "Your email",
+  "version"              : "0.0",
+  "release"              : "0.0.0",
+  "copyright"            : 2019,
+  "documentation_source" : "docs",
+  "documentation_target" : "dist",
+  "index"                : "index",
+  "language"             : "en"
```
Comment (stephenfin) [inline cookiecutter.json:14]: Not personally a fan of using this style of indentation. It doesn't look like cookiecutter supports YAML, but could we keep this as-is?

File: {{cookiecutter.project_name}}/setup.cfg [hunk 3]
```diff
@@ -6,6 +6,12 @@ description-file =
 author = {{ cookiecutter.author_name }}
 author-email = {{ cookiecutter.author_email }}
 home-page = http://www.sphinx-doc.org/
+build_sphinx=
```
Comment (stephenfin) [inline {{cookiecutter.project_name}}/setup.cfg:9]: I don't think this is a valid option for the `[metadata]` section. Rather, it should be in its own `[build_sphinx]` section.

**Comment by Carelvd** [comment]
Thank you for considering my PR.

**Pull request closed without merging**